### PR TITLE
Update all of rails to version 5.2.4.5

### PR DESCRIPTION
**Pull request by hennevogel**
`bundle update --patch --strict rails`

### CODE REVIEW

**Review by eduardoj — APPROVED** [review]
LGTM.

- Having the _Ignore "HTTP Verb Confusion" as we did in master_ commit applied, it will be convinient to backport 24f99f16e86a410a4321b4175f59c8ff52c34507 as well.
- The last commit ("Refactor validator test") should be applied to `master` branch.

**Comment by hennevogel** [comment]
> The last commit ("Refactor validator test") should be applied to master branch.

This commit is a cherry-pick from master.

**Comment by eduardoj** [comment]
> > The last commit ("Refactor validator test") should be applied to master branch.
> 
> This commit is a cherry-pick from master.

Ops, true: ce0b4aac363f33d4a648c7fdb0fcecbc252fbefd

File: src/api/lib/opensuse/validator.rb [hunk 1]
```diff
@@ -84,7 +84,7 @@ def validate(opt, content)
         if File.exist?(schema_base_filename + '.rng')
           schema = Nokogiri::XML::RelaxNG(File.open(schema_base_filename + '.rng'))
         elsif File.exist?(schema_base_filename + '.xsd')
-          schema = Nokogiri::XML::Schema(File.open(schema_base_filename + '.xsd'))
+          schema = Nokogiri::XML::Schema(File.open(schema_base_filename + '.xsd'), Nokogiri::XML::ParseOptions.new.nononet)
```
Comment (krauselukas) [inline src/api/lib/opensuse/validator.rb:87]: This change is not needed, our schemas do not include external entities, so network access for parsing is required. We are not doing it on master https://github.com/openSUSE/open-build-service/blob/50ef5823d66965dee95918f54e3d58fe70590bb4/src/api/app/lib/suse/validator.rb#L86 where we run the same version of nokogiri. It's not causing issues if we enable it, but I would stick with the defaults unless its needed

**Review by krauselukas — APPROVED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**